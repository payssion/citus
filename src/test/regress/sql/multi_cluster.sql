-- Test creation of cluster-distributed tables and metadata syncing

CREATE TABLE clustered_table (
    key text primary key,
    value jsonb
);
CREATE INDEX ON clustered_table USING GIN (value);
SELECT cluster_create_distributed_table('clustered_table', 'key', 'hash');
SELECT cluster_create_shards('clustered_table', 4);

\c - - - :worker_1_port

\d clustered_table

SELECT isowner, iscluster FROM pg_dist_partition
WHERE logicalrelid = 'clustered_table'::regclass;

SELECT count(*) FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
WHERE logicalrelid = 'clustered_table'::regclass;

\c - - - :worker_2_port

\d clustered_table

SELECT isowner, iscluster FROM pg_dist_partition
WHERE logicalrelid = 'clustered_table'::regclass;

SELECT count(*) FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
WHERE logicalrelid = 'clustered_table'::regclass;


