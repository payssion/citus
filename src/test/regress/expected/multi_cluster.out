-- Test creation of cluster-distributed tables and metadata syncing
CREATE TABLE clustered_table (
    key text primary key,
    value jsonb
);
CREATE INDEX ON clustered_table USING GIN (value);
SELECT cluster_create_distributed_table('clustered_table', 'key', 'hash');
 cluster_create_distributed_table 
----------------------------------
 
(1 row)

SELECT cluster_create_shards('clustered_table', 4);
 cluster_create_shards 
-----------------------
 
(1 row)

\c - - - :worker_1_port
\d clustered_table
Table "public.clustered_table"
 Column | Type  | Modifiers 
--------+-------+-----------
 key    | text  | not null
 value  | jsonb | 
Indexes:
    "clustered_table_pkey" PRIMARY KEY, btree (key)
    "clustered_table_value_idx" gin (value)

SELECT partmethod FROM pg_dist_partition
WHERE logicalrelid = 'clustered_table'::regclass;
 partmethod 
------------
 h
(1 row)

SELECT count(*) FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
WHERE logicalrelid = 'clustered_table'::regclass;
 count 
-------
     4
(1 row)

\c - - - :worker_2_port
\d clustered_table
Table "public.clustered_table"
 Column | Type  | Modifiers 
--------+-------+-----------
 key    | text  | not null
 value  | jsonb | 
Indexes:
    "clustered_table_pkey" PRIMARY KEY, btree (key)
    "clustered_table_value_idx" gin (value)

SELECT partmethod FROM pg_dist_partition
WHERE logicalrelid = 'clustered_table'::regclass;
 partmethod 
------------
 h
(1 row)

SELECT count(*) FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
WHERE logicalrelid = 'clustered_table'::regclass;
 count 
-------
     4
(1 row)

